src/planning/plan/plan-producer.ts

```diff
@@ -45,10 +45,11 @@ export class PlanProducer {
   searchStore?: SingletonStorageProvider;
   searchStoreCallback: ({}) => void;
   debug: boolean;
+  noSpecEx: boolean;
   inspector?: PlannerInspector;
 
-  constructor(arc: Arc, result: PlanningResult, searchStore?: SingletonStorageProvider, inspector?: PlannerInspector, {debug = false} = {}) {
-    assert(result, 'result cannot be null');                
+  constructor(arc: Arc, result: PlanningResult, searchStore?: SingletonStorageProvider, inspector?: PlannerInspector, {debug = false, noSpecEx = false} = {}) {
+    assert(result, 'result cannot be null');
```

Comment:
nit - i think a positive parameter name (i.e. `specEx` instead of `noSpecEx`) is more readable and recommended.